### PR TITLE
[libcred] Add new port

### PR DESCRIPTION
**Pull request by Shadowrom2020**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libcred/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,14 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO Shadowrom2020/libcred
+    REF baca5ae1f19d644ccb3b994f0a6530290cbb2fb2 #v0.1.0
```
Comment (Cheney-W) [inline ports/libcred/portfile.cmake:4]:
```suggestion
    REF ${VERSION}
```

File: ports/libcred/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,14 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO Shadowrom2020/libcred
+    REF baca5ae1f19d644ccb3b994f0a6530290cbb2fb2 #v0.1.0
+    SHA512 d63a51a790e3888f01efa9185e52f02ac0b49a14b15a75815b571216f1f7e825ae0e175dff41f0997595f7cbe6c6e76dc6ed357ddd58f58dc07531aba0404b05
+    HEAD_REF master
```
Comment (Cheney-W) [inline ports/libcred/portfile.cmake:6]:
```suggestion
    HEAD_REF main
```

File: ports/libcred/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    REF baca5ae1f19d644ccb3b994f0a6530290cbb2fb2 #v0.1.0
+    SHA512 d63a51a790e3888f01efa9185e52f02ac0b49a14b15a75815b571216f1f7e825ae0e175dff41f0997595f7cbe6c6e76dc6ed357ddd58f58dc07531aba0404b05
+    HEAD_REF master
+)
+
+vcpkg_cmake_configure(SOURCE_PATH ${SOURCE_PATH})
+vcpkg_cmake_install()
+
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/libcred" RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/debug/share/libcred" RENAME copyright)
```
Comment (Cheney-W) [inline ports/libcred/portfile.cmake:13]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

File: ports/libcred/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+vcpkg_cmake_config_fixup()
```
Comment (Cheney-W) [inline ports/libcred/portfile.cmake:14]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/libcred)

```

File: ports/libcred/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO Shadowrom2020/libcred
+    REF baca5ae1f19d644ccb3b994f0a6530290cbb2fb2 #v0.1.0
+    SHA512 d63a51a790e3888f01efa9185e52f02ac0b49a14b15a75815b571216f1f7e825ae0e175dff41f0997595f7cbe6c6e76dc6ed357ddd58f58dc07531aba0404b05
+    HEAD_REF master
+)
+
+vcpkg_cmake_configure(SOURCE_PATH ${SOURCE_PATH})
+vcpkg_cmake_install()
+
```
Comment (Cheney-W) [inline ports/libcred/portfile.cmake:11]:
```suggestion

   file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```

**Review by Cheney-W — CHANGES_REQUESTED** [review]

**Comment by Cheney-W** [comment]
Why don't you just use https://github.com/mamba-org/libcred directly?

**Comment by Shadowrom2020** [comment]
Good point, changed the port file to use the original repo and use meson build instead of cmake.
Sorry for the fuzz...

Thanks for the help.

________________________________
Von: Cheney Wang ***@***.***>
Gesendet: Donnerstag, 26. September 2024 11:53
An: microsoft/vcpkg ***@***.***>
Cc: Jürgen Herrmann ***@***.***>; Author ***@***.***>
Betreff: Re: [microsoft/vcpkg] [libcred] Add new port (PR #41189)


Why don't you just use https://github.com/mamba-org/libcred directly?

**Pull request closed without merging**